octodns/provider/cloudflare.py

```diff
@@ -85,11 +97,27 @@ def __init__(self, id, email=None, token=None, cdn=False, *args, **kwargs):
                 'Authorization': 'Bearer {}'.format(token),
             })
         self.cdn = cdn
+        self.retry_count = retry_count
+        self.retry_period = retry_period
         self._sess = sess
 
         self._zones = None
         self._zone_records = {}
 
+    def _try(self, *args, **kwargs):
+        tries = self.retry_count
+        while True:  # We'll raise to break after our tries expire
+            try:
+                return self._request(*args, **kwargs)
+            except CloudflareRateLimitError:
+                if tries <= 1:
+                    raise
+                tries -= 1
+                self.log.warn('rate limit encountered, pausing '
+                              'for %ds and trying again, %d remaining',
+                              self.retry_period, tries)
+                sleep(self.retry_period)
```

Comment:
Does CF return any further info in the 429 response, suggestions on back off time or anything? If so i'd be nice to utilize it, if not then a generic sleep seems fine. I see their docs say the limits are 1200/5m so guess maybe it's just a generic period and you're following what the docs say should be enough.

Reply:
No, their entire response is

```
{
	"success": false,
	"errors": [{
		"code": 10100,
		"message": "More than 1200 requests per 300 seconds reached. Please wait and consider throttling your request speed"
	}],
	"messages": [],
	"result": null
}
```

Nothing in the headers